opentreemap/treemap/udf.py

```diff
@@ -358,17 +369,24 @@ def _validate_and_update_choice(
                 {'datatype': [_("Can't change choices "
                                 "on a non-choice field")]})
 
-        if old_choice_value not in datatype['choices']:
+        choices = datatype['choices']
+        protected_choices = datatype.get('protected_choices', [])
+        all_choices = choices + protected_choices
```

Comment:
Ah, makes sense. Can we go with your suggestion before about making a helper function that takes a datatype (so we keep logic on which choices go first in one location):

```python
class UserDefinedFieldDefinition(...):
    @property
    def all_choices(self):
        return UserDefinedFieldDefinition._all_choices(self.datatype_dict)

    @classmethod
    def all_choices(cls, datatype):
        if datatype['type'] in {'choice', 'multichoice'}:
            return datatype.get('protected_choices', []) + datatype.get('choices', [])
        return None